### ISSUE TITLE
Token validation in JwtAudienceValidator failing unexpectedly

### ISSUE THREAD

**Issue by newtork** [SAP/cloud-security-xsuaa-integration#293]
Observed in version 2.5.3

The following message is logged in customer application runtime:
```
c.s.c.s.t.validation.ValidationResults   : Jwt token with audience [uaa, sb-resourceRequest-dev-1!b7469, resourceRequest-dev-1!b7469] is not issued for these clientIds: [sb-resourceRequest-dev-1!b7469, resourceRequest-dev-1!b7469].
```

I did not debug the issue, but I found the following matching source code:

![image](https://user-images.githubusercontent.com/22489773/83016357-18956e80-a022-11ea-80cb-667da0a3d3c7.png)

I do not understand how the above message can be logged. From what I can tell, `JwtAudienceValidator#validateDefault` should return a non-null value, already in the first loop iteration. What do you think? Is there something else I should check for?

**Comment by newtork** [comment]
I just realized, you didn't use the lambda notation for the else statements.
All else conditions are evaluated prematurely. This leads to misleading error messages.

**Comment by nenaraab** [comment]
hi @newtork 

thanks for reporting and all the details! We also recognized and fixed it yesterday.
https://github.com/SAP/cloud-security-xsuaa-integration/pull/290

The fix will be provided with version 2.7.3
